go-controller/pkg/clustermanager/userdefinednetwork/controller_test.go

```diff
@@ -17,535 +19,854 @@ import (
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	netv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	netv1clientset "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/client/clientset/versioned"
 	netv1fakeclientset "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/client/clientset/versioned/fake"
 	netv1informerfactory "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/client/informers/externalversions"
 	netv1Informer "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/client/informers/externalversions/k8s.cni.cncf.io/v1"
 
 	udnv1 "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/userdefinednetwork/v1"
+	udnclient "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/userdefinednetwork/v1/apis/clientset/versioned"
 	udnfakeclient "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/userdefinednetwork/v1/apis/clientset/versioned/fake"
 	udninformerfactory "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/userdefinednetwork/v1/apis/informers/externalversions"
 	udninformer "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/userdefinednetwork/v1/apis/informers/externalversions/userdefinednetwork/v1"
 
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/clustermanager/userdefinednetwork/template"
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
 )
 
 var _ = Describe("User Defined Network Controller", func() {
 	var (
-		udnClient   *udnfakeclient.Clientset
-		nadClient   *netv1fakeclientset.Clientset
-		udnInformer udninformer.UserDefinedNetworkInformer
-		nadInformer netv1Informer.NetworkAttachmentDefinitionInformer
-		kubeClient  *fake.Clientset
-		podInformer corev1informer.PodInformer
+		cs                *util.OVNClusterManagerClientset
+		udnInformer       udninformer.UserDefinedNetworkInformer
+		nadInformer       netv1Informer.NetworkAttachmentDefinitionInformer
+		podInformer       corev1informer.PodInformer
+		namespaceInformer corev1informer.NamespaceInformer
 	)
 
-	BeforeEach(func() {
-		udnClient = udnfakeclient.NewSimpleClientset()
-		udnInformer = udninformerfactory.NewSharedInformerFactory(udnClient, 15).K8s().V1().UserDefinedNetworks()
-		nadClient = netv1fakeclientset.NewSimpleClientset()
-		nadInformer = netv1informerfactory.NewSharedInformerFactory(nadClient, 15).K8sCniCncfIo().V1().NetworkAttachmentDefinitions()
+	newTestController := func(renderNADStub RenderNetAttachDefManifest, objects ...runtime.Object) *Controller {
+		cs = util.GetOVNClientset(objects...).GetClusterManagerClientset()
 
-		kubeClient = fake.NewSimpleClientset()
-		sharedInformer := informerfactory.NewSharedInformerFactoryWithOptions(kubeClient, 15)
+		udnFact := udninformerfactory.NewSharedInformerFactory(cs.UserDefinedNetworkClient, 15)
```

Comment:
the idea was to have only 1 newTestController function, and only create 1 watch factory with `factory.NewClusterManagerWatchFactory`. If you don't want to start watch factory every time, you can just make another function like `startNewTestController` that will call `newTestController` and start watch factory

Reply:
Done

Now there is single function to start the tests controller.